boefjes/boefjes/katalogus/organisations.py

```diff
@@ -10,13 +10,15 @@
 
 @router.get("", response_model=dict[str, Organisation])
 def list_organisations(storage: OrganisationStorage = Depends(get_organisations_store)):
-    return storage.get_all()
+    with storage as store:
```

Comment:
Maybe, we could implement these as a property?

Reply:
The context handler could be added in the property wrapping the list_organisations call, so that all queries inside are wrapped in a transaction.

Reply:
We could wrap these in a method, but I'd say we would have to change all of the calls here, and we would not be able to add more operations that change/search data unless we add it to the transaction in that new method and we should not call this method from inside the storage class since it could contaminate a transaction that was started inside . In other words: we would be maintaining a method per endpoint 🤔